crates/core/tedge_api/src/alarm.rs

```diff
@@ -1,58 +1,84 @@
-use std::convert::TryFrom;
-use std::convert::TryInto;
-use std::fmt;
-
+use crate::entity_store::EntityMetadata;
 use clock::Timestamp;
 use serde::Deserialize;
 use serde::Serialize;
-
 use serde_json::Value;
 use std::collections::HashMap;
-/// In-memory representation of ThinEdge JSON alarm.
-#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+use std::convert::TryFrom;
+
+/// Internal representation of ThinEdge alarm model.
+#[derive(Debug, Eq, PartialEq)]
 pub struct ThinEdgeAlarm {
```

Comment:
I really like this decomposition of an alarm in two parts: one extracted from the topic, the other from the payload.

- My initial idea was to have a single structure, but I agree that this leads to issues when parsing an MQTT message.
- Using an optional payload makes clear and less error prone the case of the empty messages used to clear alarms.

That said it would be good to find a generic approach here. The same pattern `(source, type, payload)` is used for alarms, events, measurements, commands ...